engine/server/engine/centralized_logs/client_implementations/persistent_volume/file_layout/per_hour_file_layout.go

```diff
@@ -0,0 +1,139 @@
+package file_layout
+
+import (
+	"fmt"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/logs_clock"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_consts"
+	"github.com/kurtosis-tech/kurtosis/engine/server/engine/centralized_logs/client_implementations/persistent_volume/volume_filesystem"
+	"golang.org/x/exp/slices"
+	"math"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	// basepath year/week/day/hour/
+	perHourDirPathFmtStr = "%s%s/%s/%s/%s/"
+
+	// ... enclave-uuid/service-uuid<filetype>
+	perHourFilePathFmtSt = perHourDirPathFmtStr + "%s/%s%s"
+)
+
+type PerHourFileLayout struct {
+	time             logs_clock.LogsClock
+	baseLogsFilePath string
+}
+
+func NewPerHourFileLayout(time logs_clock.LogsClock, baseLogsFilePath string) *PerHourFileLayout {
+	return &PerHourFileLayout{
+		time:             time,
+		baseLogsFilePath: baseLogsFilePath,
+	}
+}
+
+func (phf *PerHourFileLayout) GetLogFileLayoutFormat() string {
+	// Right now this format is specifically made for Vector Logs Aggregators format
+	// This wil be used my Vector LogsAggregator to determine the path to output to
+	return fmt.Sprintf("\"%s%%%%Y/%%%%V/%%%%u/%%%%H/{{ enclave_uuid }}/{{ service_uuid }}.json\"", volume_consts.LogsStorageDirpath)
+}
+
+func (phf *PerHourFileLayout) GetLogFilePath(time time.Time, enclaveUuid, serviceUuid string) string {
+	year, week, day, hour := TimeToWeekDayHour(time)
+	return phf.getHourlyLogFilePath(year, week, day, hour, enclaveUuid, serviceUuid)
+}
+
+func (phf *PerHourFileLayout) GetLogFilePaths(
+	filesystem volume_filesystem.VolumeFilesystem,
+	retentionPeriod time.Duration,
+	retentionPeriodIntervals int,
+	enclaveUuid, serviceUuid string,
+) ([]string, error) {
+	var paths []string
+	retentionPeriodInHours := DurationToHours(retentionPeriod)
+
+	if retentionPeriodIntervals < 0 {
```

Comment:
maybe could find a way to merge this `getLogFilePathsFromNowTillRetentionPeriod` and `getLogFilePathsBeyondRetentionPeriod`?